js/geo/transform.js

```diff
@@ -186,16 +179,16 @@ class Transform {
     zoomScale(zoom) { return Math.pow(2, zoom); }
     scaleZoom(scale) { return Math.log(scale) / Math.LN2; }
 
-    project(lnglat, worldSize) {
```

Comment:
Ah! Sorry about that. I think I am confusing "world size" and "tile extent" https://github.com/mapbox/mapbox-gl-js/pull/1955 in my memory. 

---

> These parameters are not related to supporting different tile sizes.

It does appear that `worldSize` has _some_ relation to `tileSize`.

```js
get worldSize() {
    return this.tileSize * this.scale;
}
```

I trust your judgement here and will ✅  the PR.